### ISSUE TITLE
Unable to configure default

### ISSUE THREAD

**Issue by realdadfish** [detekt/detekt#7559]
This is my detekt build convention:

```kotlin
plugins {
    id("io.gitlab.arturbosch.detekt")
}

detekt {
    // Bump this to receive new / adapted rules
    toolVersion = "1.23.6"
    val componentFile = rootProject.file("detekt.yml")
    val globalFile = File(getGitRoot(), "detekt.yml")
    require(componentFile.exists() || globalFile.exists()) { "No detekt.yml found" }
    val configFile = if (componentFile.exists()) componentFile else globalFile
    config.setFrom(configFile)
    buildUponDefaultConfig = true
}

tasks.withType<Detekt>().configureEach {
    // do not run detekt on generated files
    exclude { spec ->
        spec.file.absolutePath.contains("/build/")
    }
    reports {
        // html and xml are by default on, we don't need txt
        txt.required.set(false)
    }
}
```

and this my `detekt.yml` which contains this configuration:

```yml
naming:
  MatchingDeclarationName:
    active: true
    mustBeFirst: true
    multiplatformTargets:
      - 'android'
      - 'desktop'
```

I have files underneath `desktopMain` and `androidMain` named `PlatformUITest.desktop.kt` and `PlatformUITest.android.kt` respectively. Still, Detekt does not acknowledge the configuration above nor does it work with the default set of multiplatformTargets out of the box:

```
> Task :app:detektMetadataCommonMain FAILED
Property 'naming>MatchingDeclarationName>multiplatformTargets' is misspelled or does not exist.
```

**Comment by 3flex** [comment]
Which version of the plugin are you using?

**Comment by glsthomaskeller** [comment]
1.23.6 as well: `io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.23.6`

**Comment by realdadfish** [comment]
1.23.6 as well: `io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.23.6`

**Comment by realdadfish** [comment]
My bad, this seems to be unreleased still.

**Comment by cortinico** [comment]
> My bad, this seems to be unreleased still.

Was this implemented in a PR? Could you link it?

**Comment by realdadfish** [comment]
I think it was this one: https://github.com/detekt/detekt/pull/6426
